File: MobileApp/src/constants/theme.js
import {useContext} from 'react';
import {Platform} from 'react-native';
import {ThemeContext} from '../contexts/ThemeContext';

/**
 * Modern Theme Colors
 * Dark mode first approach with light mode toggle
 */

// Dark theme colors (default) - Elegant champagne gold theme
const DARK_COLORS = {
  // Primary - Champagne gold accent
  primary: '#C8B588',
  primaryDark: '#B09D6F',
  primaryLight: '#D8C38F',

  // Backgrounds - Soft black & charcoal
  background: '#0C0C0C',
  secondaryBackground: '#1A1A1A',
  groupedBackground: '#0C0C0C',
  secondaryGroupedBackground: '#1A1A1A',
  cardBackground: '#1A1A1A',
  tertiaryBackground: '#242424',

  // Text - High contrast, elegant
  text: '#FAFAFA',           // Headings
  textSecondary: '#CCCCCC',   // Body text
  textTertiary: '#888888',    // Muted text
  placeholder: '#666666',

  // Status
  success: '#D8C38F',         // Light champagne for highlights
  error: '#EF4444',
  warning: '#F59E0B',
  info: '#6A7A8C',           // Ink-blue accent

  // UI Elements
  separator: 'rgba(255, 255, 255, 0.06)',
  border: 'rgba(255, 255, 255, 0.08)',
  link: '#6A7A8C',           // Ink-blue accent

  // Buttons & Controls
  buttonFill: 'rgba(255, 255, 255, 0.06)',
  secondaryButtonFill: '#1A1A1A',
};

// Light theme colors - Elegant light version
const LIGHT_COLORS = {
  // Primary - Champagne gold accent
  primary: '#B09D6F',
  primaryDark: '#8F7D52',
  primaryLight: '#C8B588',

  // Backgrounds - Soft whites and creams
  background: '#FAFAFA',
  secondaryBackground: '#FFFFFF',
  groupedBackground: '#FAFAFA',
  secondaryGroupedBackground: '#FFFFFF',
  cardBackground: '#FFFFFF',
  tertiaryBackground: '#F5F5F5',

  // Text - Dark, elegant
  text: '#0C0C0C',
  textSecondary: '#333333',
  textTertiary: '#666666',
  placeholder: '#999999',

  // Status
  success: '#B09D6F',
  error: '#EF4444',
  warning: '#F59E0B',
  info: '#6A7A8C',

  // UI Elements
  separator: 'rgba(0, 0, 0, 0.08)',
  border: 'rgba(0, 0, 0, 0.12)',
  link: '#6A7A8C',

  // Buttons & Controls
  buttonFill: 'rgba(0, 0, 0, 0.04)',
  secondaryButtonFill: '#F5F5F5',
};

// Hook to get current theme colors
export const useThemeColors = () => {
  const themeContext = useContext(ThemeContext);
  // If ThemeContext is not available (e.g., during initialization), default to dark mode
  const isDarkMode = themeContext?.isDarkMode ?? true;
  return isDarkMode ? DARK_COLORS : LIGHT_COLORS;
};

// iOS Typography Scale (Dynamic Type)
export const FONT_SIZES = {
  largeTitle: 34,    // iOS Large Title
  title1: 28,        // iOS Title 1
  title2: 22,        // iOS Title 2
  title3: 20,        // iOS Title 3
  headline: 17,      // iOS Headline (semibold)
  body: 17,          // iOS Body (regular)
  callout: 16,       // iOS Callout
  subheadline: 15,   // iOS Subheadline
  footnote: 13,      // iOS Footnote
  caption1: 12,      // iOS Caption 1
  caption2: 11,      // iOS Caption 2
};

// iOS Font Weights
export const FONT_WEIGHTS = {
  ultraLight: '100',
  thin: '200',
  light: '300',
  regular: '400',
  medium: '500',
  semibold: '600',
  bold: '700',
  heavy: '800',
  black: '900',
};

// iOS Standard Font Families
export const FONTS = {
  // Use SF Pro (iOS system font) - React Native uses by default
  regular: {
    fontFamily: Platform.OS === 'ios' ? 'System' : 'Roboto',
    fontWeight: '400',
  },
  medium: {
    fontFamily: Platform.OS === 'ios' ? 'System' : 'Roboto',
    fontWeight: '500',
  },
  semibold: {
    fontFamily: Platform.OS === 'ios' ? 'System' : 'Roboto',
    fontWeight: '600',
  },
  bold: {
    fontFamily: Platform.OS === 'ios' ? 'System' : 'Roboto',
    fontWeight: '700',
  },
};

// iOS Standard Spacing (8pt grid)
export const SPACING = {
  xxs: 2,
  xs: 4,
  sm: 8,
  md: 16,    // Standard margin
  lg: 20,
  xl: 24,
  xxl: 32,
  xxxl: 48,
};

// iOS Standard Border Radius
export const BORDER_RADIUS = {
  xs: 4,
  sm: 6,
  md: 8,
  lg: 10,
  xl: 12,
  xxl: 16,
  round: 999,
};

// iOS Standard Shadows
export const SHADOWS = {
  small: {
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 1},
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1,
  },
  medium: {
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.20,
    shadowRadius: 3.0,
    elevation: 2,
  },
  large: {
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 4},
    shadowOpacity: 0.22,
    shadowRadius: 5.0,
    elevation: 4,
  },
};

// iOS Standard Component Heights
export const HEIGHTS = {
  navBar: 44,
  tabBar: 49,
  listRow: 44,
  listRowLarge: 60,
  button: 44,
  input: 44,
};

// iOS Standard Layout
export const LAYOUT = {
  screenPadding: SPACING.md,        // 16pt margin from edges
  listInset: SPACING.md,            // 16pt list inset
  sectionSpacing: SPACING.lg,       // 20pt between sections
  cardPadding: SPACING.md,          // 16pt inside cards
  listSeparatorInset: SPACING.md,   // 16pt separator inset
};

// Animation durations (iOS standard)
export const ANIMATION = {
  quick: 200,
  standard: 300,
  slow: 500,
};

File: iOSApp/VintedNotifications/Constants/Theme.swift
//
//  Theme.swift
//  Vinted Notifications
//
//  Modern theme with dark/light mode support
//  Elegant champagne gold theme
//

import SwiftUI

// MARK: - Theme Colors

struct AppColors {
    // Primary - Champagne gold accent
    let primary: Color
    let primaryDark: Color
    let primaryLight: Color

    // Backgrounds
    let background: Color
    let secondaryBackground: Color
    let groupedBackground: Color
    let secondaryGroupedBackground: Color
    let cardBackground: Color
    let tertiaryBackground: Color

    // Text
    let text: Color
    let textSecondary: Color
    let textTertiary: Color
    let placeholder: Color

    // Status
    let success: Color
    let error: Color
    let warning: Color
    let info: Color

    // UI Elements
    let separator: Color
    let border: Color
    let link: Color

    // Buttons & Controls
    let buttonFill: Color
    let secondaryButtonFill: Color
}

// Dark theme colors (default)
let darkColors = AppColors(
    // Primary - Champagne gold accent
    primary: Color(hex: "C8B588"),
    primaryDark: Color(hex: "B09D6F"),
    primaryLight: Color(hex: "D8C38F"),

    // Backgrounds - Soft black & charcoal
    background: Color(hex: "0C0C0C"),
    secondaryBackground: Color(hex: "1A1A1A"),
    groupedBackground: Color(hex: "0C0C0C"),
    secondaryGroupedBackground: Color(hex: "1A1A1A"),
    cardBackground: Color(hex: "1A1A1A"),
    tertiaryBackground: Color(hex: "242424"),

    // Text - High contrast, elegant
    text: Color(hex: "FAFAFA"),
    textSecondary: Color(hex: "CCCCCC"),
    textTertiary: Color(hex: "888888"),
    placeholder: Color(hex: "666666"),

    // Status
    success: Color(hex: "D8C38F"),
    error: Color(hex: "EF4444"),
    warning: Color(hex: "F59E0B"),
    info: Color(hex: "6A7A8C"),

    // UI Elements
    separator: Color.white.opacity(0.06),
    border: Color.white.opacity(0.08),
    link: Color(hex: "6A7A8C"),

    // Buttons & Controls
    buttonFill: Color.white.opacity(0.06),
    secondaryButtonFill: Color(hex: "1A1A1A")
)

// Light theme colors
let lightColors = AppColors(
    // Primary - Champagne gold accent
    primary: Color(hex: "B09D6F"),
    primaryDark: Color(hex: "8F7D52"),
    primaryLight: Color(hex: "C8B588"),

    // Backgrounds - Soft whites and creams
    background: Color(hex: "FAFAFA"),
    secondaryBackground: Color(hex: "FFFFFF"),
    groupedBackground: Color(hex: "FAFAFA"),
    secondaryGroupedBackground: Color(hex: "FFFFFF"),
    cardBackground: Color(hex: "FFFFFF"),
    tertiaryBackground: Color(hex: "F5F5F5"),

    // Text - Dark, elegant
    text: Color(hex: "0C0C0C"),
    textSecondary: Color(hex: "333333"),
    textTertiary: Color(hex: "666666"),
    placeholder: Color(hex: "999999"),

    // Status
    success: Color(hex: "B09D6F"),
    error: Color(hex: "EF4444"),
    warning: Color(hex: "F59E0B"),
    info: Color(hex: "6A7A8C"),

    // UI Elements
    separator: Color.black.opacity(0.08),
    border: Color.black.opacity(0.12),
    link: Color(hex: "6A7A8C"),

    // Buttons & Controls
    buttonFill: Color.black.opacity(0.04),
    secondaryButtonFill: Color(hex: "F5F5F5")
)

// MARK: - Font Sizes (iOS Typography Scale)

struct FontSizes {
    static let largeTitle: CGFloat = 34    // iOS Large Title
    static let title1: CGFloat = 28        // iOS Title 1
    static let title2: CGFloat = 22        // iOS Title 2
    static let title3: CGFloat = 20        // iOS Title 3
    static let headline: CGFloat = 17      // iOS Headline (semibold)
    static let body: CGFloat = 17          // iOS Body (regular)
    static let callout: CGFloat = 16       // iOS Callout
    static let subheadline: CGFloat = 15   // iOS Subheadline
    static let footnote: CGFloat = 13      // iOS Footnote
    static let caption1: CGFloat = 12      // iOS Caption 1
    static let caption2: CGFloat = 11      // iOS Caption 2
}

// MARK: - Spacing (8pt grid)

struct Spacing {
    static let xxs: CGFloat = 2
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 16    // Standard margin
    static let lg: CGFloat = 20
    static let xl: CGFloat = 24
    static let xxl: CGFloat = 32
    static let xxxl: CGFloat = 48
}

// MARK: - Border Radius

struct BorderRadius {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 6
    static let md: CGFloat = 8
    static let lg: CGFloat = 10
    static let xl: CGFloat = 12
    static let xxl: CGFloat = 16
    static let round: CGFloat = 999
}

// MARK: - Shadows

struct Shadows {
    static let small = (color: Color.black.opacity(0.18), radius: CGFloat(1.0), x: CGFloat(0), y: CGFloat(1))
    static let medium = (color: Color.black.opacity(0.20), radius: CGFloat(3.0), x: CGFloat(0), y: CGFloat(2))
    static let large = (color: Color.black.opacity(0.22), radius: CGFloat(5.0), x: CGFloat(0), y: CGFloat(4))
}

// MARK: - Heights

struct Heights {
    static let navBar: CGFloat = 44
    static let tabBar: CGFloat = 49
    static let listRow: CGFloat = 44
    static let listRowLarge: CGFloat = 60
    static let button: CGFloat = 44
    static let input: CGFloat = 44
}

// MARK: - Layout

struct Layout {
    static let screenPadding: CGFloat = Spacing.md       // 16pt margin from edges
    static let listInset: CGFloat = Spacing.md          // 16pt list inset
    static let sectionSpacing: CGFloat = Spacing.lg     // 20pt between sections
    static let cardPadding: CGFloat = Spacing.md        // 16pt inside cards
    static let listSeparatorInset: CGFloat = Spacing.md // 16pt separator inset
}

// MARK: - Animation

struct Animation {
    static let quick: Double = 0.2
    static let standard: Double = 0.3
    static let slow: Double = 0.5
}

// MARK: - Color Extension for Hex Support

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }

        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue:  Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

// MARK: - Theme Environment Key

struct ThemeKey: EnvironmentKey {
    static let defaultValue: AppColors = darkColors
}

extension EnvironmentValues {
    var theme: AppColors {
        get { self[ThemeKey.self] }
        set { self[ThemeKey.self] = newValue }
    }
}

// MARK: - Theme Manager

class ThemeManager: ObservableObject {
    @Published var isDarkMode: Bool = true

    var currentTheme: AppColors {
        isDarkMode ? darkColors : lightColors
    }

    func toggleTheme() {
        isDarkMode.toggle()
    }
}
